sklearn/discriminant_analysis.py

```diff
@@ -254,6 +255,18 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+    @property
+    def covariance_(self):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
```

Comment:
So now I realise that the former behaviour was correct, and this warning is not necessary or appropriate. Sorry.